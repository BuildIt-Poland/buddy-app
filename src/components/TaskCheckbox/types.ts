import { TaskStatus } from 'buddy-app-schema';

export interface CheckboxProps {
  onChange?: (id: string) => void;
}
export interface TaskCheckboxProps extends CheckboxProps {
  id: string;
  title?: string;
  status?: TaskStatus;
}
