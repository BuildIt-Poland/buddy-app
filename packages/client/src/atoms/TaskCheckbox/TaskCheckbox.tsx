import React from 'react';
import { TaskStatus } from '@buddy-app/schema';
import { makeStyles } from '@material-ui/core/styles';
import Checkbox from '@material-ui/core/Checkbox';
import UncheckedCircleIcon from '@material-ui/icons/RadioButtonUnchecked';
import CheckedCircleIcon from '@material-ui/icons/CheckCircle';
import { TaskCheckboxProps } from './types';

const useStyles = makeStyles(() => ({
  large: {
    fontSize: '3.25rem',
  },
  small: {
    fontSize: '2.25rem',
  },
}));

const TaskCheckbox: React.FC<TaskCheckboxProps> = ({
  task,
  size = 'default',
  onChange,
  hasRipple,
  edge,
}) => {
  const { large, small } = useStyles();
  const isChecked = task.status === TaskStatus.Completed;
  const sizeClassNames = {
    inherit: '',
    default: '',
    large,
    small,
  };

  const onCheckboxChange = () => onChange && onChange(task);

  return (
    <Checkbox
      color={'primary'}
      checked={isChecked}
      onChange={onCheckboxChange}
      disableRipple={!hasRipple}
      edge={edge}
      icon={<UncheckedCircleIcon className={sizeClassNames[size]} />}
      checkedIcon={<CheckedCircleIcon className={sizeClassNames[size]} />}
    />
  );
};

export default TaskCheckbox;
