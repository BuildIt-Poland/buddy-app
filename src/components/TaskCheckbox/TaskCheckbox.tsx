import React from 'react';
import Checkbox from '@material-ui/core/Checkbox';
import { TaskStatus } from 'buddy-app-schema';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import UncheckedCircleIcon from '@material-ui/icons/RadioButtonUnchecked';
import CheckedCircleIcon from '@material-ui/icons/CheckCircle';
import { TaskCheckboxProps } from './types';

const TaskCheckbox: React.FC<TaskCheckboxProps> = ({
  id,
  className,
  title,
  status,
  onChange,
}) => {
  const isChecked = status === TaskStatus.Completed;
  const text = {
    [TaskStatus.Completed]: <del>{title}</del>,
    [TaskStatus.Uncompleted]: <strong>{title}</strong>,
  };
  const onCheckboxChange = () => onChange && onChange(id);

  const renderCheckbox = () => (
    <Checkbox
      className={className}
      color={'primary'}
      checked={isChecked}
      onChange={onCheckboxChange}
      disableRipple
      onClick={e => e.stopPropagation()}
      icon={<UncheckedCircleIcon />}
      checkedIcon={<CheckedCircleIcon />}
    />
  );

  return title ? (
    <ListItem button>
      <ListItemIcon>{renderCheckbox()}</ListItemIcon>
      <ListItemText primary={text[status]} />
    </ListItem>
  ) : (
    renderCheckbox()
  );
};

export default TaskCheckbox;
