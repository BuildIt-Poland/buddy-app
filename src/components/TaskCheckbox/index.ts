import TaskCheckbox from './TaskCheckbox';
export * from './types';

export default TaskCheckbox;
