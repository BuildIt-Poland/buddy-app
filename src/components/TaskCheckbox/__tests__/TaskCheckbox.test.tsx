import React from 'react';
import { create } from 'react-test-renderer';

import TaskCheckbox from '../TaskCheckbox';

jest.mock('@material-ui/core/Box', () => 'Box');

describe('Component - TaskCheckbox', () => {
  test('renders correctly', () => {
    const component = create(<TaskCheckbox />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
