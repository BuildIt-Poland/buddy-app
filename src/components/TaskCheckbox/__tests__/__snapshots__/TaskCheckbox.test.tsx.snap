// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TaskCheckbox renders correctly when task is COMPLETED 1`] = `
<Box
  className="makeStyles-wrapper-1"
>
  <Checkbox
    checked={true}
    color="primary"
    inputProps={
      Object {
        "aria-label": "primary checkbox",
      }
    }
    onChange={[Function]}
    value="1"
  />
  <Box
    className="makeStyles-titleWrapper-2"
  >
    <Typography
      component="h4"
      variant="body1"
    >
      <Box
        fontWeight="fontWeightRegular"
      >
        Change you line manager
      </Box>
      <div
        className="makeStyles-line-3"
      />
    </Typography>
  </Box>
</Box>
`;

exports[`Component - TaskCheckbox renders correctly when task is UNCOMPLETED 1`] = `
<Box
  className="makeStyles-wrapper-1"
>
  <Checkbox
    checked={false}
    color="primary"
    inputProps={
      Object {
        "aria-label": "primary checkbox",
      }
    }
    onChange={[Function]}
    value="1"
  />
  <Box
    className="makeStyles-titleWrapper-2"
  >
    <Typography
      component="h4"
      variant="body1"
    >
      <Box
        fontWeight="fontWeightBold"
      >
        Change you line manager
      </Box>
    </Typography>
  </Box>
</Box>
`;
